### DISCUSSION TITLE
Project Optic Package

### DISCUSSION THREAD

**Discussion by Paulumvi** [3b1b/manim#1504] [General]
I have just started diving into Manim. I would like to illustrate some of my projects in the field of optics. To do this I have to write  objects like lenses and methods like beam propagation. Is it beneficial to do this here as a public project, both to share it with everyone and may to get some feedback or support on my code?

**Comment by ghost** [comment]
Beam propogation in manim is not that good.

manim is made for e, exponentials and the complex numberline, not split-quaternions or Quasi for short.

Quasi × Quaternion × Quasi should get you started on efficient beam propogation, just use HSL colors and not RGB.

Hue (Complex), Saturation (Quasi) and Luminosity (Real) is the solution.

See if you can write the code for efficient beam propogation with texture, think fourier.
